### PR TITLE
Update references to k8s 1.6 and older for calico 2.5 KDD

### PR DESCRIPTION
**Pull request by gunjan5**
## Description

- Calico v2.5+ with Kubernetes backend will only support Kubernetes v1.7+ 
- This PR also removes extra CRDs from the installation manifests that are not needed for policy-only mode ( CC @tmjd) 

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.7/calico.yaml [hunk 1]
```diff
@@ -204,54 +205,6 @@ spec:
 ---
 
 apiVersion: apiextensions.k8s.io/v1beta1
-description: Calico BGP Peers
```
Comment (ozdanborne) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.7/calico.yaml]: I don't have the background for why this is removed...

File: v2.5/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md [hunk 2]
```diff
@@ -65,21 +66,14 @@ networking with a full node-to-node mesh and/or explicit configuration of peers.
 To install Calico with Calico networking, run one of the commands below based on your Kubernetes version.
 This will install Calico and will initially create a full node-to-node mesh.
 
-For **Kubernetes 1.6+** clusters:
+> Note: Calico `v2.5.0` or higher only support Kubernetes `v1.7.0` or higher.
```
Comment (ozdanborne) [inline v2.5/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md]: typo: `supports`

I also think the phrasing could be improved a little:

Note: Calico v2.5 with Kubernetes-Backend requires Kubernetes v1.7+.

?

Comment (gunjan5) [inline v2.5/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md]: yeah that sounds better, I'll update it

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.7/calico.yaml (hunk 1 above)
Comment (gunjan5) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.7/calico.yaml]: these CRDs are not needed for policy-only mode

**Pull request merged**